iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -51,6 +52,130 @@ inline FixedPositionContainer<T, CAPACITY>::~FixedPositionContainer() noexcept
     }
 }
 
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(const FixedPositionContainer& rhs) noexcept
+{
+    *this = rhs;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(FixedPositionContainer&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(const FixedPositionContainer& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = Index::FIRST;
+        ConstIterator rhs_it = rhs.begin();
+
+        // copy using copy assignment util rhs iter becomes invalid
+        for (; rhs_it.to_index() != Index::INVALID; i++, rhs_it++)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                // clean original data by it's dtor
+                erase(i);
+            }
+            m_data[i] = *rhs_it;
```

Comment:
Using the copy assignment after deletion is undefined behavior. You can use copy assignment when the slot is used and placement new when it is free

Reply:
If I understand correctly, this would fix the problem:

```c
// assign directly, no need to call erase (all memory located on stack)
m_data[i] = (m_status[i] == SlotStatus::USED) ? (*rhs_it) : new(&m_data[i]) T(*rhs_it);
```

Reply:
Hahaha, yes. I had to think a bit to realize that it works. Interesting solution but for readability reasons an if-else statement would be better.

Reply:
Sure :)

Reply:
See https://github.com/eclipse-iceoryx/iceoryx/commit/e28c2d798c936d69fc90e82d37abaf50d6b97017.